core/src/main/java/org/apache/accumulo/core/client/impl/AbstractId.java

```diff
@@ -39,7 +39,7 @@ protected AbstractId(final String canonical) {
   /**
    * The canonical ID
    */
-  public final String canonicalID() {
+  public String canonicalID() {
```

Comment:
Why make this non-final? Given how important this method is for its serialization/internal representation, it's pretty important that it not be overridden.

Reply:
Definitely an oversight on my part, fixed.

Reply:
Ah, good. I was worried that this was required for the serialization to work for some reason.